### ISSUE TITLE
Familairize and document on wiki the integration of RPG-NPC features into the current source code

### ISSUE THREAD

**Issue by philipchenaux** [uchicago-cs/chiventure#436]
I plan to look into the existing source code to both familiarize myself generally with how the code is set-up, but at the same time understand and roughly plan how NPC features can be tied in terms of code. As I will be working as part of the RPG-NPC team, I'll be specifically for opportunities where such features can be integrated. Unfortunately, this is quite vague to start with, but as I look into it, hopefully I'll be able to update this issue with more concrete and actionable takeaways. I'll be working closely with @avianurag, who will look at the NPC dependencies across other teams to come up with a plan on NPC feature integration.

**Comment by philipchenaux** [comment]
Briefly looked through the code, given the focus on the rpg aspects of npc features a lot of the implementation of such features could potentially be within the actions module.

**Comment by philipchenaux** [comment]
Created wiki documentation covering the integration of rpg-npc features into the current chiventure framework. I also included our team's dependencies and overlaps with other teams which allows for further discussion with other teams to flesh out what different aspects each of us should cover. Below are the initial details I added:

> Familiarization with Chiventure modules and opportunities for rpg-npc implementation
npc battle feature

Aspects such as npc battle features would work closely with the wdl module to integrate the possibility of a battle object. Moreover the actions module would need to be built upon to include battle actions. We may also integrate the possibility of saving and loading the current game state within a fight which would require work with the game_state module.
Team Dependencies
NPC Team

One aspect we are naturally relying upon is the struct that the NPC team would create is a npc module wherein there is an npc struct containing the general characteristics of each npc.
